src/lxc/commands.c

```diff
@@ -147,10 +147,12 @@ static int lxc_cmd_rsp_recv(int sock, struct lxc_cmd_rr *cmd)
 
 		rspdata = malloc(sizeof(*rspdata));
 		if (!rspdata) {
-			ERROR("Failed to allocate response buffer for command \"%s\"",
-			      lxc_cmd_str(cmd->req.cmd));
-			return -ENOMEM;
+			errno = ENOMEM;
+			SYSERROR("Failed to allocate response buffer for command \"%s\"",
```

Comment:
The logging calls will clobber `errno` again. I think returning the value is fine in this function btw., as long as we do it consistently (see below), given that the callers want to handle `ECONNRESET` specifically, so the style guide would currently prefer returning a negative error code.
Also I think the error message is detailed enough and since we know the exact text the `errno` will produce (since we're assigning it), I'm not sure it's worth switching to the `SYS*` macro.

Reply:
The answer is to not globber errno in the log. I'll push a series of patches on top of this one that will take care of this.